Assets #597 (#611)
plotly/dash-core-components: README.md

# Dash Core Components

This package provides the core React components suite for [Dash][].

[![CircleCI](https://circleci.com/gh/plotly/dash-core-components.svg?style=svg)](https://circleci.com/gh/plotly/dash-core-components)

## Development

### Testing Locally

The `dash` package contains some tools to build components and drive the bundles build process.

> To avoid the circulair dependency situation, we don't add dash as a required install in dash-core-components setup. But in order to do the local development, you need to install `dash` before everything.

Reviewer: ```suggestion
To avoid the circular dependency situation, we don't add `dash` as a required install in the `dash-core-components` setup. But, in order to development locally, you need to install `dash` before everything.
```

# Dash Core Components

This package provides the core React components suite for [Dash][].

[![CircleCI](https://circleci.com/gh/plotly/dash-core-components.svg?style=svg)](https://circleci.com/gh/plotly/dash-core-components)

## Development

The `dash` package contains some tools to build components and drive the bundles build process.

> To avoid the circular dependency situation, we don't add `dash` as a required install in the `dash-core-components` setup. But, in order to do development locally, you need to install `dash` before everything.

Reviewer 2: :cool:

```bash
# it's recommended to install your python packages in a virtualenv
# python 2
$ pip install virtualenv --user && virtualenv venv && . venv/bin/activate
# python 3
$ python -m venv && . venv/bin/activate

# make sure dash is installed with dev and testing dependencies
$ pip install dash[dev,testing]  # in some shells you need \ to escape []

# run the build process with build
$ npm run build  # or `python scripts/build.py build`

# install dcc in editable mode
$ pip install -e .
```

### The local build process

> Dash-core-components nows shares the same build process as in dash-renderer. The build

Reviewer 2: ```suggestion
`dash-core-components` now shares the same build process as `dash-renderer`. The build
```

> tool can be called by `npm run build` or `python scripts/build.py build`.  There are several sub commands in `build.py`, they are explained in [dash-renderer guide](https://github.com/plotly/dash/blob/dev/CONTRIBUTING.md#dash-renderer-beginner-guide).

Reviewer 2: ```suggestion
tool can be called by `npm run build` or `python scripts/build.py build`.  There are several sub-commands in `build.py`; they are explained in the [dash-renderer guide](https://github.com/plotly/dash/blob/dev/CONTRIBUTING.md#dash-renderer-beginner-guide).
```

You have to maintain the list of components in `demo/Demo.react.js`.

### Linting and Testing

> Note: the integration test needs an extra selenium setup, please refer to [install section](https://dash.plot.ly/testing)

Reviewer 2: ```suggestion
Note: the integration test requires Selenium to be set up, please refer to [install section](https://dash.plot.ly/testing)
```

> You can run everything in one command by `npm run test`. There are also a list of test command available inside `scripts` property of `package.json` file.

Reviewer 2: ```suggestion
You can run everything in one command by `npm run test`. There are also a list of test command available inside the `scripts` property of `package.json` file.
```
